aptos-move/framework/aptos-framework/sources/keyless_account.move

```diff
@@ -98,81 +111,136 @@ module aptos_framework::keyless_account {
         }
     }
 
-    // Sets the Groth16 verification key, only callable via governance proposal.
-    // WARNING: If a malicious key is set, this would lead to stolen funds.
-    public fun update_groth16_verification_key(fx: &signer, vk: Groth16VerificationKey) acquires Groth16VerificationKey {
-        system_addresses::assert_aptos_framework(fx);
-
-        if (exists<Groth16VerificationKey>(signer::address_of(fx))) {
-            let Groth16VerificationKey {
-                alpha_g1: _,
-                beta_g2: _,
-                gamma_g2: _,
-                delta_g2: _,
-                gamma_abc_g1: _
-            } = move_from<Groth16VerificationKey>(signer::address_of(fx));
+    /// Pre-validate the VK to actively-prevent incorrect VKs from being set on-chain.
+    fun validate_groth16_vk(vk: &Groth16VerificationKey) {
+        // Could be leveraged to speed up the VM deserialization of the VK by 2x, since it can assume the points are valid.
+        assert!(option::is_some(&crypto_algebra::deserialize<bn254_algebra::G1, bn254_algebra::FormatG1Compr>(&vk.alpha_g1)), E_INVALID_BN254_G1_SERIALIZATION);
```

Comment:
if move bn254 is disabled, we can't update vk?

Reply:
Yep. Is that a problem? I figured it would be wise to check validity here & preempt bad VKs from being set.